stdlib/public/core/HashedCollections.swift.gyb

```diff
@@ -2439,36 +2425,77 @@ collections = [
 ]
 }%
 
+% for (Self, a_self, TypeParametersDecl, TypeParameters, AnyTypeParameters, Sequence, AnySequenceType) in collections:
+
+#if _runtime(_ObjC)
+extension ${Self} {
+  /// Return the pointer to the underlying storage buffer, force-cast
+  /// to an integer. Only used for unit tests.
+  public func _identifierForStorageBuffer() -> Int {
+    switch _variantStorage {
+    case let .native(storage):
+      return unsafeBitCast(storage.buffer, to: Int.self)
+    case let .cocoa(storage):
+      return unsafeBitCast(storage.cocoa${Self}, to: Int.self)
+    }
+  }
+}
+#else
+extension ${Self} {
+  /// Return the pointer to the underlying native storage buffer, force-cast
+  /// to an integer. Only used for unit tests.
+  public func _identifierForStorageBuffer() -> Int {
+    return unsafeBitCast(_variantStorage.asNative.buffer, to: Int.self)
+  }
+}
+#endif
```

Comment:
This extension can be moved to the unit test itself, I think.  You can just bitcast the `self` value.